internal/repos/syncer.go

```diff
@@ -616,10 +616,27 @@ func (s *Syncer) SyncExternalService(
 	interval := calcSyncInterval(now, svc.LastSyncAt, minSyncInterval, modified, errs)
 
 	s.log().Debug("Synced external service", "id", externalServiceID, "backoff duration", interval)
+
+	// Re-load the service from the DB and update the sync timestamps. We do this
+	// because a sync can take a long time and we'd potentially over-write changes to
+	// other fields like the config that happened during the sync by re-using the same `svc`
+	// loaded at the beginning of the sync.
+	tx, err := s.Store.ExternalServiceStore.Transact(ctx)
+	if err != nil {
+		return errors.Append(errs, errors.Wrap(err, "failed to start tx"))
+	}
+
+	defer func() { errs = errors.Append(errs, tx.Done(err)) }()
```

Comment:
@tsenart This line looks fishy to me. We're appending to `errs` on defer, but `errs` is not the return error value. Basically, I think this is ignoring any errors returned from `tx.Done()`.

Reply:
Edit: I see further down, we're doing `return errs`, which is implicitly setting `err = errs`, which I _think_ is fine, but I'm struggling to follow the flow of errors in this chunk

Reply:
I think this would surface any errors:
```suggestion
	defer func() { err = errors.Append(err, tx.Done(err)) }()
```